src/Avalonia.Controls/SelectionNode.cs

```diff
@@ -28,6 +29,7 @@ internal class SelectionNode : IDisposable
         private readonly SelectionNode? _parent;
         private readonly List<IndexRange> _selected = new List<IndexRange>();
         private readonly List<int> _selectedIndicesCached = new List<int>();
+        private SelectionModelChangeSet? _changes;
```

Comment:
nit: I wonder if this would be a little more simplified if these were refactored in a separate class by themselves. There is some duplication between _selected and keeping track of the changeset.

Reply:
Hmm, could you explain more? I'm struggling to see the duplication.